indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -189,14 +188,27 @@
 
 (def REINDEX_BATCH_SIZE 2000)
 
+(defn- remove-nil-tail
+  "Remove any nils from the end of a collection."
+  [coll]
+  (->> coll
```

Comment:
reverse the collection twice to drop items from the end does not pass the smell test. Change it to something like:
```
(loop [x coll] 
  (if (last x) 
    x 
    (recur (drop-last x))))
```